metaflow/plugins/kfp/kfp.py

```diff
@@ -735,7 +736,10 @@ def step_op(
                     preceding_component_inputs=preceding_component_inputs,
                     preceding_component_outputs=preceding_component_outputs,
                 ),
-                base_image=self.base_image,
+                base_image=kfp_component.kfp_decorator.attributes["image"]
+                if kfp_component.kfp_decorator
+                and kfp_component.kfp_decorator.attributes["image"]
```

Comment:
style: could you please put parentheses for the conditional? It'd be easier to read, or create an `image` variable.

Reply:
Do you mean something like `if (kfp_component.kfp_decorator and kfp_component.kfp_decorator.attributes["image"])`?